[Op#52605]: Add Loading state to Project Storage OAuth Grant Access nudge modal

akabiru opened the pull request:
### https://community.openproject.org/wp/52605

On "Login" click, transition to loading state; render back on success (failure state to be handle in [#Bug/52798](https://community.openproject.org/wp/52798))

- [x] Update the "Login" button to a form button to preseve semantic A11ly focus & tab benefits. The previous implementaion with a "button link" breaks accessibility as ScreenReader can't tab to it. Also buttons have the advantage of executing via KYBD `Enter` or `Spacebar`
- [x]  On modal load, set focus on the "Login" button as the default action. The user can exit easily via `ESC` or tab back to the two close actions: Close button at the top OR the cancel "I will do it later" button.
- [x] Define loading state as an [`aria-live`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) region to enable screen readers to announce loading status 

Accessibility References:

_Key A11ly AC:_ Convey when login has started, while it's in progress and when it's completed or failed

* https://primer.style/ui-patterns/loading#accessibility
* https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role
* https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions


https://github.com/opf/openproject/assets/17295175/3d8ea32d-92a9-4090-b761-0b96f62e621c


https://github.com/opf/openproject/assets/17295175/6fa46438-dd42-4b51-aa70-e7fcd2fc9424

akabiru committed:
chore[Op#52605]: Target "Login" button click, prevent default start loading screen

* Update the "Login" button to a form button to preseve semantic A11ly
  focus & tab benefits. The previous implementaion with a "button link"
  breaks accessibility as ScreenReader can't tab to it. Also buttons have
  the advantage of executing via KYBD `Enter` or `Spacebar`
* On modal load, set focus on the "Login" button as the default action.
  The user can exit easily via `ESC` or tab back to the two close
  actions: Close button at the top OR the cancel "I will do it later"
  button.

Accessibility References:

* https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

akabiru committed:
feat[Op#52605: redefine waiting modal body title and subtitle as arguments

akabiru committed:
feat[Op#52605]: Define the loading state modal

akabiru committed:
a11y[Op#52605]: Define aria live POLITENESS_SETTING for nudge modal

akabiru committed:
chore[Op#52605]: normalize i18n file

akabiru committed:
fix[Op#52605]: Update close button label value & target

akabiru committed:
tests[Op#52605]: Amend modal interface to allow overrides for preview

akabiru committed:
tests[Op#52605]: update specs

akabiru committed:
a11y[Op#52605]: Update the aria labels

akabiru committed:
tests[Op#52605]: Include requests specs for completeness

akabiru committed:
tests[Op#52605]: fixup test label

akabiru committed:
a11y[Op#52605]: Convey loading process to screen readers

akabiru committed:
a11y[Op#52605]: Announce access granted message as first result

ba1ash reviewed (changes requested):
I played with it locally and that is what I found:
-  I faced a translation problem problem. 
![2024-02-26_12-32](https://github.com/opf/openproject/assets/11717656/28197b2c-25db-4538-8082-b794ea0e2ad2)

   I assume it is due to https://github.com/opf/openproject/blob/57ec904bbd9182d1b0617b2e6184862a852fdb7c/modules/storages/app/components/storages/admin/oauth_access_grant_nudge_modal_component.rb#L35 being resolved too early. Probably it should be a component method.
- In Chromium I face `Refused to send form data to 'https://openproject.local/projects/30/settings/project_storages/87/oauth_access_grant?utf8=%E2%9C%93' because it violates the following Content Security Policy directive: "form-action 'self' openproject-development.s3.amazonaws.com openproject-development.s3.eu-west-1.amazonaws.com".` after hitting `Login`.
  See warning in https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/form-action. Can we use `window.location.href =` to send user to do login?
- Somehow `Do it later` and cross sign does not work in the initial modal. Is that intentional? I click and nothing happens. See video, please.
- In Chromium I see a modal jump from the top to the center of the page. Probably, we should try to avoid it. See video, please.
-  I wonder if we should initialize nudge modal on storage edit as well. Wdyt?


https://github.com/opf/openproject/assets/11717656/a459558c-33a5-4e83-a2ca-031231ea10f9

akabiru committed:
fix[Op#52605]: Do not use options dynamic methods that do not change
state

When `I18n.locale` is changed at runtime, the default option API does
not work as it "caches" the state

Co-authored-by: Pavel Balashou <p.balashou@openproject.com>

akabiru committed:
fix[Op#52605]: Do not use contested requestSubmit form action

Automated requestAccessFormTarget.requestSubmit() is not possible due to CSP restrictions
See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/form-action

Co-Authored-by: Pavel Balashou <p.balashou@openproject.com>

akabiru committed:
tests[Op#52605]: fix undefined method `role` for `Capybara::Node::Simple`

akabiru committed:
tests[Op#52605]: update feature tests to assert against final storage path

ba1ash reviewed (approved):
Great work, Kaburu! :+1:
